### DISCUSSION TITLE
Aggiornamento nameserver dyndns per cloudfare

### DISCUSSION THREAD

**Discussion by Borghese** [indomus/forum#2687] [HUB: Home Assistant]
Ciao a tutti
ho un dominio registrato con dyndns.org
Sto attivando cloudfare per poi configurare l'accesso remoto a home assistant, ma a un certo punto della procedura, dovrei impostare su dyndns i dns che mi fornisce cloudfare
Non trovo questa opzione da nessuna parte sul loro sito, qualcuno ha idea di come fare ?

**Comment by snakuzzo** [comment]
Sei sicuro si possa fare con dyndns.org ? Non è che sia matematimante possibile con qualsiasi registrar

**Comment by Borghese** [comment]
ah ecco.....eh mi sa di no allora... che p@@@@@
ce l'ho attivo da una vita, e volevo sfruttarlo (visto che lo pago) per configurare accesso remoto su ha attraverso cloudfare
Quindi a sto punto devo cambiare registrar....

alternativa consigliata è godaddy vero ?

**Reply by snakuzzo** [reply]
noi abbiamo testato godaddy (per la guida che trovi sul sito). Io personalmente ho comprato su namecheap.

**Reply by snakuzzo** [reply]
Continui a rispondere creando nuovi thread...devi fare REPLY sotto l'ultimo messaggio!!! NON SO PIU' COME DIRVELO... CAXXO!!! :)

**Reply by Borghese** [reply]
azzo scusaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa

mi inganna il fatto che la finestra più grande sotto è quella per creare nuovi thread! 
Oh e sì che non sono proprio novizio sui forum eh.... mi pesto le mani promesso scussa!

**Reply by snakuzzo** [reply]
Sì, ma questo di fatto non è un forum. Lo utilizziamo come tale, ma non lo è

**Reply by snakuzzo** [reply]
e vedrai che appena ti bannerò non ti confonderai più :D

**Reply by Borghese** [reply]
finite le cartucce, recepito il messaggio :(

AWS dici che è sprecato per questo utilizzo vero ?
Ho dato un occhiata ai prezzi di godaddy e mamma mia... dopo il primo anno schizza...

Ora mi faccio un giro

**Reply by snakuzzo** [reply]
Guarda che il prezzo dipende anche dal dominio che scegli...e comunque al massimo spendi 10-15€/anno...non credo che sia un prezzo così alto, visti anche i vantaggi che avrai

**Reply by Borghese** [reply]
si l'unico sarebbe un .it a circa 16€ l'anno.....

**Reply by Borghese** [reply]
bon.... ho acquistato un dominio.me a 11,7€ e poi va a 28€
Tra l'altro mi sono reso conto che a dyndns davo la modica cifra di 55$ l'anno per un servizio di ddns..... ho chiuso tutto!

**Reply by Borghese** [reply]
Ok ho fatto tutto seguendo la guida! grazie @snakuzzo !
Unico problema è che non mi parte la mail di verifica da cloudflare, quindi non posso autorizzare il tunnel.... non me ne va bene una!!!

**Reply by Borghese** [reply]
nulla da fare ragazzi, non riesco ad accedere da app su smartphone a home assistant
Su cloudflare il mio dominio è attivo
ho sostituito i dns con i loro nameserver su godaddy
Il tunnel è autorizzato e attivo
ma se imposto sull'app come nome server : https://dominiocreato.me:8123 non si connette

mi dice che il server non esiste...

avete suggerimentio

**Reply by Borghese** [reply]
aggiornamento : Se su app imposto il mio ip pubblico:8123, raggiungo l'istanza di home assistant
Se metto il nome dominio invece no. è come se non venisse risolto il dns a sto punto....
Eppure su cloudflare mi sembra tutto ok... da cosa può dipenderE?

**Reply by Borghese** [reply]
a stro punto mi vien da pensare che godaddy non abbia ancora aggiornato i  dns.... ma se cosi fosse cloudflare non mi darebbe il dominio attivo giusto ?

**Reply by snakuzzo** [reply]
non ti seguo

**Reply by Borghese** [reply]
Scusa non volevo spammare il thread, ma davvero non riesco a venirne a capo.
Ho seguito alla lettera le guide [quì ](https://indomus.it/guide/collegarsi-da-remoto-a-home-assistant-via-reti-provider-tradizionali-o-che-usino-nat-via-cloudflare/#conclusionil)e [quì](https://indomus.it/guide/configurare-cloudflare-per-il-proprio-dominio-personale-per-protezione-reverse-proxy-cdn-e-altro/) ma non riesco a raggiungere home assistant dall'app.

Ho attivato il dominio con godaddy (un dominio .me)
Ho attivato il dominio su cloudflare, sostituendo i nameserver a quelli di godaddy
Su cloudflare è tutto attivo
Il tunnel con cloudflare da home assistant, è attivo e dai log non vedo errori
ma se imposto sull'app il server https://miodominio.me non lo raggiungo, mi dice che non esiste
Ho provato anche con https://miodominio.me:8123 (la porta è forwardata sul router vs ip raspberry che ospita istanza di home assistant) ma nulla da fare
Se invece, sia da app che da browser, punto a : ippubblico:8123 lo raggiungo senza problemi
quindi è qualcosa che non risolve i dns... ma non so più cosa controllare....

**Reply by snakuzzo** [reply]
Hai letto la guida linkata su come passare da duckdns a cloudflared ?

- Devi innanzi tutto avere HA configurato in http
- Assicurarti che il tunnel sia correttamente configurato (sulla dashboard di Cloudflare)
- puoi tranquillamente rimuovere le regole di Port forwarding perché non ti serviranno più (visto che non esporrai nulla direttamente su internet)
- verifica i log del servizio Cloudflare

Più di questo non so dirti...ed è inutile continuare a scrivere che non riesci linkando le guide che hai seguito. Per chi legge diventa difficile. Al massimo condividi screenshot, configurazioni, log... E non "io ho fatto questo, questo e quest'altro.

**Reply by Borghese** [reply]
Hai ragione, provo a rispondere per punti : 

1) è una configurazione ex novo, quindi non usavo duckdns,
2) Questa è la dashboard di cloudflare, da cui sembra che il dominio e i dns siano attivi : 

![a](https://github.com/indomus/forum/assets/5764547/aba08409-0040-4ef6-981c-7ff91cb01778)
![b](https://github.com/indomus/forum/assets/5764547/ef39e1ff-badb-4cfa-a91c-056c44e2e5aa)

3) Questi invece sono i nameserver di cloudflare inseriti nel mio dominio godaddy : 

![c](https://github.com/indomus/forum/assets/5764547/e2d8852d-cdc8-4f83-9a4f-6c9b2567574e)

 4) Questo è come è configurato l'http sul file configuration.yaml su ha : 

![d](https://github.com/indomus/forum/assets/5764547/ce283b5f-3bf7-4305-baee-05e4c64a4fe4)

5) Questo invece è il registro dell'add on di cloudflare su home assistant, dove mi pare che il tunnel sia stabilito e non ci siano errori : 

![e](https://github.com/indomus/forum/assets/5764547/36a7a8a3-4e77-48e5-b0fa-c185c834282b)

il dominio che ho creato è myhometech.me

**Reply by Borghese** [reply]
intanto ho fatto qualche altro test, per capire  meglio : 

Ho notato che il mio dominio (inteso come pagina web del sito) di godaddy non era raggiungibile su web, poichè i dns (di cloudflare) non venivano risolti.
Allora ho provato a rimettere i nameserver di godaddy, rimuovendo quelli di cloudflare
Tempo mezz'ora e la mia pagina web è tornata su : www.myhometech.me

A me il problema che ho postato sopra sembra un ritardo nella propagazione dei nameserver tra godaddy e cloudflare, che ne pensate ?
Anche se di fatto erano passate quasi 24h....

Se non risolvo ahimè devo tornare alla configurazione precedente, con un servizio di ddns e esponendo la porta 8123 con forward... non ho altre soluzioni.

p.s domanda forse banale ma su web non ho trovato nulla : Ma godaddy fornisce un servizio di ddns ? cioè quando il provider mi cambia ip dinamico, godaddy come fa a saperlo ?

**Reply by snakuzzo** [reply]
Scusa, però per queste domande non fai prima a cercare direttamente sul sito di godaddy ?
Comunque...il tuo approccio non mi piace...perché il discorso tunnel funziona a prescindere. Se hai problemi non è che risolvi tornando indietro e buttando tutto. Se ci sono problemi lato cloudflare aspetti e fine.
Tra le altre cose, se leggi i log dell'addon, dovresti anche aggiornarlo.
Ad ogni modo, dai log e da quello che vedo sulla dashboard di Cloudflare il tunnel è stato creato correttamente per `myhometech.me` e funziona tutto correttamente.

**Reply by Borghese** [reply]
Niente da fare alla fine ho  dovuto desistere : 

Il mio provider (sky wifi) usa il MAP-T per la condivisione dell'ip pubblico (ipv4 1/16) quindi cloudflare e godaddy non riuscivano a indirizzare e risolvere la richiesta.
Ho dovuto configurare tutto con duckdns e porta 8123 forwardata sul router
Peccato...

grazie cmq a tutti e scusate per lo spam, magari sarà utile a qualcun altro nella mia situazione

**Reply by snakuzzo** [reply]
Mi sa che è na caxxata... questo sistema serve proprio per queste situazioni in cui si è nattati.
Cloudflare e Godaddy non devono contattare il tuo IP pubblico, ma è esattamente il contrario!
Il tunnel viene stabilito dall'addon e non il contrario...quindi non trarre conclusioni affrettate e senza senso.

**Reply by Borghese** [reply]
@snakuzzo per favore, non fraintendere : il mio approccio non è di certo avere la pappa pronta, ma piuttosto sfruttare queste occasioni per imparare, e magari lasciare traccia sul forum se qualcuno in futuro incappa nella mia stessa situazione.
Ho voluto comunque ripercorrere i passi all'indietro, con i dns etc etc, proprio per capire dove si incastrava il processo, ma per es nel mio caso, una volta aggiornati i nameserver su godaddy con quelli di cloudflare, tempo 30 min che propagassero, e il mio dominio risultava irragiungibile. Nonostante il tunnel tra add on e cloudflare funzionasse correttamente.
Poi chattando con assistenza tecnica, è emerso il tema del MAP-T

**Reply by Borghese** [reply]
no aspetta, ma questa cosa del map-t me l'ha detta l'assistenza in chat di godaddy... dici che è una caxxata ?!?

**Reply by snakuzzo** [reply]
sì, a mio avviso il tizio ti ha detto una cazzata. O lui non ha capito bene cosa stai facendo, oppure tu non gliel'hai spiegato correttamente. Dico questo perché sei tu (dal tuo PC) a contattare Cloudflare per creare il tunnel...e inoltre, come ti ho già detto, se guardi i log e la dashboard di cloudflare, funziona tutto correttamente. Il tunnel è stato creato...fine.
Vista tale situazione, mi spieghi per quale cacchio di motivo godaddy dovrebbe contattare il tuo ip pubblico ? Ed ecco il motivo per cui ti ho scritto che trai conclusioni affrettate.
 
Quanto all'irraggiungibilità del tuo dominio, potrebbe tranquillamente dipendere da altro.
Forse ti sfugge qualcosa...
Nel momento in cui cambi i nameserver per passare il dominio sotto cloudflare, che cosa ti aspetteresti di trovare ?
Poi...che vuol dire irraggiungibile ? Qual è il problema ? Non viene risolto ? Viene risolto e dà un errore Cloudflare ? Viene risolto e dà altro errore ? Se non specificate bene cosa accade, nessuno potrà dire mai cosa è successo.

**Reply by snakuzzo** [reply]
Ma poi dico...che cacchio c'entra la chat di godaddy quando ormai il dominio è in gestione sotto Cloudflare ?

**Reply by Borghese** [reply]
Si ma tu hai ragione sul funzionamento "tecnico" del processo, in effetti non fa una piega! 
La cosa "strana" e che non sono riuscito a capire è questa, ma che a sto punto mi ha tratto in inganno, è questa : 
Attivando il dominio con godaddy, ho attivato anche una pagina web, che rispondeva ovviamente al nome di dominio. E fin quì, tutto ok, la raggiungevo anche tramite web.
Nel momento in cui cambiavo i nameserver con quelli di cloudflare, tempo mezz'ora e la pagina web non  era più raggiungibile.
Il browser ti diceva che non riusciva a risolvere i dns (err net resolve una cosa del genere)
se pingavo il dominio la richiesta scadeva
idem tracert e nslookup
Come se quel dominio non esistesse.

Eppure... su cloudflare era tutto attivo, e il tunnel funzionava.

Possibile a sto punto che fosse solo un problema di "tempo di propagazione" ? magari serviva più tempo non lo so...

**Reply by Borghese** [reply]
Si ma ci posso riprovare eh, quando una cosa non funziona mi incaponisco se non capisco il perchè... e tu alla fine hai ragione a cazziarmi :)
Lo rifaccio con piùà calma, e aspetto un paio di gg per capire se è una questione di tempo di propagazione

**Reply by snakuzzo** [reply]
perdonami di nuovo...ma che diavolo c'entra quella pagina web di godaddy ? Capisci che stai andando a testare una cosa che non devi testare ?
Ci fai una sega con traceroute, non ti serve assolutamente a nulla.
Attiva sto benedetto tunnel e poi fai le prove...NON PRIMA!!!

**Reply by Borghese** [reply]
Va bene! Rifaccio tutto e vi aggiorno! grazie @snakuzzo
